
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
    <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
    <head>
    	<!-- updated 2016-01-25 -->
    	
        <!-- site meta data -->  
        <meta http-equiv="content-type" content="text/html; charset=utf-8" />
        <meta name="description" content="NeXML is a file format for encoding phyloinformatic data using XML." />
        <meta name="keywords" content="NeXML, XML, NEXUS, phyloinformatics, phylogenetics, phylogenies, trees, dna, rna, nucleotide, standard, continuous, amino, acid, matrix, tree, network, data" />
        <meta name="author" content="Rutger Vos" />
        <meta name="robots" content="index,follow" />
        <meta name="verify-v1" content="mYceuNu58e0c0TNJ/gJ2NO63bxeH1nu28sZ2rtzcLoM=" />
                
        <title>NeXML - phylogenetic data as xml</title>  
        <link rel="stylesheet" href="/css/style.css" type="text/css" />
        <link rel="stylesheet" href="/css/schema.css" type="text/css" />
        <script type="text/javascript" src="/js/functions.js"></script>
        <script type="text/javascript" src="/js/schema.js"></script>
        <link rel="shortcut icon" href="/img/favicon.ico" />
    </head>
    <body>
    
    <!-- This heading is only here for text browsers without css -->
    <h1 class="hide">NeXML - phylogenetic data as xml</h1>
    
    <div id="thetop">
      <a id="top" name="top"></a>
      <p class="hide">
        Skip to: 
        <a href="#sitemenu" accesskey="2">Site menu</a> |
        <a href="#maincontent" accesskey="3">Main content</a>
      </p>
    </div>
    
    <div id="container">
      <div id="main">

        <!-- main banner in the top-left -->
        <div id="logo">       
          <h1>
            [<a href="/" accesskey="4">NeXML</a>]
          </h1><span id="tagline">Rich phyloinformatic data</span>

        </div>
        
        <!-- site blurb in the top-middle -->
        <div id="intro">
          <h2>
            <a id="maincontent" name="maincontent"></a>
            The future data exchange standard is here!
          </h2>
          <p>
            NeXML is an exchange standard for representing 
            phyloinformatic data &mdash; inspired by the commonly used 
            NEXUS format, but more robust and easier to process.
          </p>
        </div>
        <div class="clear"></div>    
        
        <!-- validate form -->   
        <div id="validateDiv">
            <form 
                action="/phylows/validator" 
                enctype="multipart/form-data" 
                method="post" 
                id="validateForm">
                <fieldset id="validateFieldset">
                    <legend class="hide">Process nexml data</legend>
                    <select id="validateSelect">
                    	<option value="/phylows/validator">validate</option>
                    	<option value="/phylows/nex2xml">nexus -> nexml</option>
                    	<option value="/phylows/newick2xml">newick -> nexml</option>
                    	<option value="/phylows/xml2json">nexml -> json</option>
                    	<option value="/phylows/xml2rdf">nexml -> rdf</option>
                    </select>
                    <input type="file" name="file" id="validateUpload"/>
                    <input type="button" onclick="submitForm()" id="validateSubmit" value="Submit"/>
                </fieldset>
            </form>          
        </div>   
        
        <!-- main heading of this page -->
        <h3 class="headerstyle">Overview</h3>
        
        	<!-- http://www.nexml.org/ -->
                <div style="padding-bottom:1em;margin-top: 1em; margin-left: 10px;color: #888888">
            <small></small>
        </div>        
                                
		<!-- TITLE: NeXML - phylogenetic data as xml -->
		<!-- HEADING: Overview -->
		<p>
			The <a rel="nofollow" href="http://www.citeulike.org/user/rvosa/article/2011773">
			NEXUS flat file format</a> is a commonly used syntax for
			phylogenetic data. Unfortunately, over time, non-compliant NEXUS implementations
			have overloaded the standard - which has caused various
			<a rel="nofollow" href="https://www.nescent.org/wg_evoinfo/NEXUS_Problems">
			problems</a>. Meanwhile, mature technologies around the
			<a rel="nofollow" href="http://www.w3.org/XML/">XML</a> standard have
			emerged. These technologies have the potential to greatly
			simplify and improve robustness in the processing of rich phylogenetic
			data. This website is the home for the community-driven NeXML project, which
			seeks to leverage XML technologies in the development of a data standard
			that translates NEXUS concepts into a syntax that is more easily validated
			and processed. This approach promises several advantages:
		</p>
		<ul>
		<li class="rss">
			<strong>Syntax validation</strong> &mdash; some of the issues hampering
			interoperability are caused by the fact that no formal
			specification exists for NEXUS and other flat files, and no unambiguous way
			to validate them. Using <a rel="nofollow" href="http://www.w3.org/XML/Schema">
			XML Schema</a> we have defined a versioned grammar against which
			data files can be validated syntactically. In addition, this website has a validation
			service (the orange box in the center of every page) that also checks the semantics
			of uploaded NeXML files beyond ways that can be expressed in XSD schema language.
		</li>
		<li class="rss">
			<strong>Semantic annotation</strong> &mdash; an issue in current file
			formats is that their semantics are not well-defined. For example, what does
			it mean to use an ambiguity code in a matrix? Is it uncertainty or polymorphism?
			With the wider <a rel="nofollow" href="https://www.nescent.org/wg_evoinfo/">
			EvoInfo working group</a> we are developing an <a href="http://www.evolutionaryontology.org">
			ontology</a> on which we are mapping NeXML schema types so that the semantics
			of data files become well-defined. In addition, NeXML has a facility for
			annotating fundamental phylogenetic data objects (such as trees, character
			state matrices and taxa) with ontology predicates and objects using
			<a href="http://www.w3.org/TR/xhtml-rdfa-primer/">RDFa</a>.
		</li>
		<li class="rss">
			<strong>Web services</strong> &mdash; a number of different
			technologies (such as <a rel="nofollow" href="http://www.xmlrpc.com/">XML-RPC</a>,
			<a rel="nofollow" href="http://en.wikipedia.org/wiki/Representational_State_Transfer">REST</a>
			and <a rel="nofollow" href="http://www.w3.org/TR/soap/">SOAP</a>) have
			emerged allowing disparate, xml-based services to be glued together
			over the internet. For example, the
			<a href="http://evoinfo.nescent.org/PhyloWS" rel="nofollow">PhyloWS</a>
			initiative seeks to develop conventions for RESTful phylogenetic web
			services for which NeXML is one of the preferred response formats.
		</li>
		<!--li class="rss">
			<strong>Native XML databases</strong> &mdash; relational
			databases are based on a fixed schema. For phylogenetic
			data this poses problems, because the field of phyloinformatics
			moves very rapidly: new metrics and analysis types are
			published constantly.
			<a rel="nofollow" href="http://xmldb-org.sourceforge.net/">XML databases</a>
			are freed from this constraint, simplifying storage of
			unusual data types.
		</li-->
		</ul>
		<p>
			Because of the advantages of NeXML over current standards, developers of phylogenetic 
			software have come together as part of the <a href="http://www.nescent.org">
			NESCent</a> working group for <a href="https://www.nescent.org/wg_evoinfo">
			evolutionary informatics</a> to develop this
			<a href="https://github.com/nexml/nexml/wiki/NeXML-Manual">
			new data exchange standard</a> based on XML technologies.
		</p>
		<p>
			<strong>We have recently published a description of NeXML in <em>Systematic
			Biology</em>. If you use NeXML in your research, please consider citing
			<a href="http://dx.doi.org/10.1093/sysbio/sys025">doi:10.1093/sysbio/sys025</a></strong>
		</p>
		<p class="internallink">

		  [ <a href="#top">Back to top</a> ]
		</p>
		<h3 class="headerstyle">
			What are we doing about it?
		</h3>
		<p>
			NeXML development is being undertaken along a number of tracks:
		</p>
		<ul>
			<li class="rss">
				In the first place, there's an XML schema. This
				schema (currently designated as namespace http://www.nexml.org/2009)
				is explained on our
				<a href="https://github.com/nexml/nexml/wiki/NeXML-Manual">wiki</a> and
				<a href="/doc/schema-1">formally documented</a>; the latest version is available from
				<a href="https://github.com/nexml/nexml/tree/master/xsd">git</a>.
			</li>
			<li class="rss">
				Secondly, the community is implementing NeXML read and/or write abilities in a
				number of software applications:<br/>
				<table id="implementations">
					<tr>
						<td>
							<img alt="rOpenSci" src="/img/rOpenSci.png"/>
						</td>
						<td>
              Carl Boettiger and Scott Chamberlain have developed an
              <a
                href="https://github.com/ropensci/RNeXML">excellent NeXML
                library for R</a> for
              <a
                href="http://ropensci.org/">rOpenSci</a>.
						</td>
					</tr>
					<tr>
						<td>
							<img alt="TreeBASE" src="/img/treebase.jpg"/>
						</td>
						<td>
							<a href="http://www.treebase.org">
							TreeBASE</a> supports serialization to NeXML.<br/>
						</td>
					</tr>			
					<tr>
						<td>
							<img alt="Mesquite" src="/img/mesquite.gif"/>
						</td>
						<td>
							The <a href="http://mesquiteproject.org">Mesquite</a> project
							supports <a href="https://github.com/nexml/nexml.java">reading
							and writing</a> of NeXML. Wayne Maddison and Peter Midford
							helped start an implementation for this that is currently being
							maintained by Rutger Vos.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="DAMBE" src="/img/dambe.gif"/>
						</td>
						<td>			        
							Xuhua Xia's <a href="http://dambe.bio.uottawa.ca/dambe.asp">DAMBE</a>
							version 5.2.31 for Windows Vista/7 reads and writes NeXML
							data.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="PhenoScape" src="/img/phenoscape_logo.jpg"/>
						</td>
						<td>
							The <a href="http://www.phenoscape.org">PhenoScape</a> 
							project uses NeXML to annotate complex morphological character
							states with ontology terms in its
							<a href="http://phenoscape.org/wiki/Phenex">Phenex editor</a>.<br/>
						</td>
					</tr>			
					<!--tr>
						<td>
							<img alt="R" src="/img/Rlogo.jpg"/>
						</td>
						<td>
							The <a href="https://r-forge.r-project.org/projects/phylobase/">
							phylobase</a> package for <a rel="nofollow" href="http://www.r-project.org/">
							R</a> reads and writes tree descriptions, with character matrices under way. 
							This implementation is being developed by Aaron Mackey.<br/>
						</td>
					</tr-->
					<tr>
						<td>
							<img alt="DendroPy" src="/img/dendropy1.png"/>
						</td>
						<td>
							Jeet Sukumaran has implemented NeXML I/O for python in the 
							<a href="https://dendropy.org/">DendroPy</a> package.
							There are many DendroPy
							<a href="https://github.com/nexml/nexml/wiki/NeXML-Manual">code samples</a>
							for dealing with NeXML data in the wiki manual.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="BioPerl" src="/img/BioPerlLogo.png"/>
						</td>
						<td>
							Chase Miller has implemented
							<a href="http://doc.bioperl.org/releases/bioperl-current/bioperl-live/Bio/NexmlIO.html">Bio::NexmlIO</a> for
							<a href="http://www.bioperl.org">BioPerl</a>, which under
							the hood reuse Rutger Vos's
							<a href="http://search.cpan.org/dist/Bio-Phylo/">Bio::Phylo</a>
							parser libraries.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="BioRuby" src="/img/bioruby-gem.png"/>
						</td>
						<td>
							Anurag Priyam and Rutger Vos have developed a 
							<a href="http://rubygems.org/gems/bio-nexml">NeXML I/O plugin</a>
							for the
							<a href="http://www.bioruby.org">
							BioRuby</a> open source bioinformatics library for Ruby.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="ETE" src="/img/ete_logo.png"/>
						</td>
						<td>
							Jaime Huerta-Cepas' team is working on NeXML I/O for the
							<a href="http://etetoolkit.org">
							ETE</a> Python environment for tree exploration.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="mx" src="/img/mx_icon.png"/>
						</td>
						<td>
							Matt Yoder has implemented NeXML serialization for the
							<a href="http://mx.phenomix.org/index.php/Main_Page">
							mx</a> collaborative web-based content management system for evolutionary 
							systematists.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="PhyloBox" src="/img/phylobox.jpg"/>
						</td>
						<td>
							Andrew Hill has added NeXML support to 
							<a href="http://phylobox.appspot.com/">
							PhyloBox</a>.<br/>
						</td>
					</tr>			
					<tr>
						<td>
							<img alt="jsPhyloSVG" src="/img/jsphylosvg.png"/>
						</td>
						<td>
							Sam Smits has made it so that the 
							<a href="http://www.jsphylosvg.com">
							jsPhyloSVG</a> tree visualization widget can now show NeXML trees.<br/>
						</td>
					</tr>
					<tr>
						<td>
							<img alt="NamesOnNodes" src="/img/namesonnodes.png"/>
						</td>
						<td>
							Mike Keesey has added NeXML support to <a href="http://namesonnodes.org/">Names On Nodes</a>,
							a web application that automatically applies biological nomenclature to datasets.
						</td>
					</tr>
					<tr>
						<td>
							<img alt="DendroScope" src="/img/dendroscope.jpg"/>
						</td>
						<td>
							Daniel Huson's <a href="http://ab.inf.uni-tuebingen.de/software/dendroscope/">DendroScope</a>
							development team has adopted NeXML as its primary file format for storing visualization (styling)
							metadata. However, at present, its implementation is <strong>not</strong> fully
							compliant so should not be taken as a template to generate NeXML output.
						</td>
					</tr>			

					<tr>
						<td>
							<img alt="phyloGeoRef" src="/img/phylogeoref.jpg"/>
						</td>
						<td>
							For the 2011 Google Summer of Code, Apurv Verma has added
							NeXML reading capability to
							<a href="https://github.com/kdiverson/phyloGeoRef">phyloGeoRef</a>.<br/>
						</td>
					</tr>						
					<tr>
						<td>
							<img alt="HIVQuery" src="/img/hivq.jpg"/>
						</td>
						<td>
							Mark Jensen has implemented NeXML compatability for the
							<a href="http://fortinbras.us/">HIVQuery</a> web
							application.<br/>
						</td>
					</tr>			
				</table>
			</li>
			<li class="rss">
				Third, we're crossreferencing the NeXML schema with the
				<a rel="nofollow" href="http://www.evolutionaryontology.org">Character Data Analysis Ontology</a>
				which is being developed by other members of the EvoInfo working group.
			</li>
		</ul>
		<p class="internallink">
		  [ <a href="#top">Back to top</a> ]
		</p>
		<h3 class="headerstyle">
		  Get involved!
		</h3>
		<p>
		  If you are interested in being involved in the NeXML project
		  in any way, please do! Here are some ways to get involved:
		</p>
		<ul>
			<li class="rss">
				<strong>Get informed</strong> &mdash; information about the NeXML
				project is distributed over the <a href="https://github.com/nexml/nexml/wiki/NeXML-Manual">manual</a>
				(for an overview of vision, plans, implementation),
				<a href="/doc/schema-1">documentation</a>
				(for formal description of the schema) and the
				<a href="https://lists.sourceforge.net/lists/listinfo/nexml-discuss">mailing list</a>
				(for immediate plans and discussion).
			</li>
			<li class="rss">
				<strong>Try it out</strong> &mdash; the download section of the
				website has nightly builds of bindings for various languages.
				Take these for a spin!
			</li>
			<li class="rss">
				<strong>Contribute</strong> &mdash; if you are a programmer interested
				in extending NeXML support, please contact us through the mailing
				list to get commit support for the subversion repository.
			</li>
		</ul>
		<p class="internallink">
		  [ <a href="#top">Back to top</a> ]
		</p>
		<h3 class="headerstyle">
			Acknowledgements
		</h3>
		<p>
			The research leading to these results has received funding from the [European
			Community's] Seventh Framework Programme ([FP7/2007-2013] under grant agreement
			n° [237046].
		</p>
		<p class="internallink">
		  [ <a href="#top">Back to top</a> ]
		</p>
		<h3 class="headerstyle">
			Suggested citation
		</h3>
		<p>
			<strong>Rutger A. Vos, J. P. Balhoff, J. A. Caravas, M. T. Holder, H. Lapp, P. E. 
			Midford, A. Priyam, J. Sukumaran, X. Xia, and A. Stoltzfus</strong>. 2012. NeXML: 
			rich, extensible, and verifiable representation of comparative data and metadata. 
			<em>Systematic Biology</em> <strong>61</strong>(4): 675-689 
			[<a href="http://dx.doi.org/10.1093/sysbio/sys025">doi:10.1093/sysbio/sys025</a>]
		</p>
		<p class="internallink">
		  [ <a href="#top">Back to top</a> ]
		</p>
      </div>
      
      <div id="sidebar">
      
        <!-- top-right site navigation menu -->
        <h2 class="quicklinks">
          <a id="quicklinks" name="quicklinks"></a>Quick links:
        </h2>
          <a class="quicklink" href="https://github.com/nexml/nexml/wiki/NeXML-Manual">Manual</a>
          <a class="quicklink" href="/doc/schema-1">Schema documentation</a>
          <a class="quicklink" href="http://dx.doi.org/10.1093/sysbio/sys025">NeXML publication</a>
          <a class="quicklink" href="https://github.com/nexml/nexml/tree/master/examples">Example files</a>  
          <a class="quicklink" href="http://www.slideshare.net/rvosa/nexml-11158998">Slide show</a>      
        <br />
        <h2 class="sidelink menuheader">
          <a name="sitemenu"></a>Libraries:
        </h2>
        
        <a class="sidelink" href="https://github.com/nexml/nexml.java">Java</a> 
        <span class="hide">|</span>
        
        <a class="sidelink" href="http://www.dendropy.org/">Python</a>
        <span class="hide">|</span> 
        
        <a class="sidelink" href="http://search.cpan.org/dist/Bio-Phylo/">Perl</a> 
        <span class="hide">|</span>
        
		<a class="sidelink" href="https://github.com/nexml/nexml.cpp">C++</a> 
        <span class="hide">|</span>		
        
        <a class="sidelink" href="https://github.com/nexml/nexml.js">JavaScript</a> 
        <span class="hide">|</span> 
        
        <a class="sidelink" href="https://github.com/nexml/bio-nexml">Ruby</a> 
        <span class="hide">|</span>
           
        
        <a class="hide" href="#top" accesskey="1">Top of page</a>
        
        <!-- links at the bottom right -->
        <h3>
          External links
        </h3>
        <ul>        
          <li class="external">
          	<a href="https://github.com/nexml" rel="nofollow">nexml@github</a>
          </li>
          <li class="external">
          	<a href="https://github.com/nexml/nexml/wiki/NeXML-Manual">NeXML wiki</a>
          </li>
          <li class="external">
          	<a href="http://www.evolutionaryontology.org/" rel="nofollow">CDAO</a>
          </li>
		  <li class="external">
			<a href="http://evoinfo.nescent.org/PhyloWS" rel="nofollow">PhyloWS</a>
		  </li>
        </ul>
      </div>
      <div class="clear">
        &nbsp;
      </div>
    </div>
    <div id="footer">
      <p>
        Site implementation and technology:
        &copy; <a href="http://rutgervos.blogspot.com" class="credit">Rutger Vos</a>. 
        Site design: &copy; <a href="http://andreasviklund.com/" class="credit">Andreas Viklund</a> 
        of <a href="http://jokkmokk.biz/" title="ITUS Jokkmokk">Jokkmokk</a>.
      </p>
    </div>
    <!--
	    <script src="http://www.google-analytics.com/urchin.js" type="text/javascript">
	    </script>
	    <script type="text/javascript">
	    _uacct = "UA-3174308-1";
	    urchinTracker();
	    </script>  
    -->  
  </body>
</html>
